File: Try_GameEngine5/src/com/example/try_gameengine/utils/IDetectAreaRequest.java
package com.example.try_gameengine.utils;

/**
 * {@code IDetectAreaRequest} is an wrap of DetectArea which use to request.
 * @author irons
 *
 */
public interface IDetectAreaRequest {
	public void setDetectArea(DetectArea request);
	public DetectArea getDetectArea();
	public String getTag();
	public void setTag(String tag);
	public Object getObjectTag();
	public void setObjectTag(Object objectTag);
}

File: Try_GameEngine5/src/com/example/try_gameengine/utils/IDetectAreaRequest.kt
package com.example.try_gameengine.utils

/**
 * `IDetectAreaRequest` is an wrap of DetectArea which use to request.
 * @author irons
 // */
interface IDetectAreaRequest {
    fun getDetectArea(): DetectArea
    fun setDetectArea(detectArea: DetectArea)
    fun getTag(): String?
    fun setTag(tag: String?)
    fun getObjectTag(): Any?
    fun setObjectTag(objectTag: Any?)
}
